### PR TITLE
fix(deis): validate the syntax of the environment file

### LINKED ISSUE #3658: deis config:pull and config:push fail on empty lines
Whenever an empty line is present in the .env file (like at the end), the two commands ``deis config:pull`` and ``deis config:push`` fail in a cryptic way.

Just skip empty lines, please.

    $ deis --version
    1.6.0

It behaves like this:

    [cimnine@local myapp]$ cat .env
    bla
    
    [cimnine@local myapp]$ deis config:push
    Usage: deis config:push [options]
    [cimnine@local myapp]$ deis config:pull
    Traceback (most recent call last):
      File "<string>", line 2466, in <module>
      File "<string>", line 2462, in main
      File "<string>", line 2395, in _dispatch_cmd
      File "<string>", line 1332, in config_pull
    IndexError: list index out of range

### PR DESCRIPTION
**Pull request by kalbasit**
fixes #3658

### CODE REVIEW

**Comment by eMxyzptlk** [comment]
Note that this still does not allow multi-line config and does not validate them.

File: client/deis.py [hunk 1]
```diff
@@ -1302,6 +1302,19 @@ def config_unset(self, args):
...
 
+    def _read_config_from_path(self, path):
+        env_dict = {}
+        with open(path, 'r') as f:
+            for line in f.readlines():
+                line = line.strip()
+                if re.match(r'.+=.+', line) is None:
+                    self._logger.warning('could not parse config line: "%s"', line)
+                    continue
+                k, v = line.split('=', 1)[0], line.split('=', 1)[1]
```
Comment (mboersma) [inline client/deis.py]: This can be simplified to:
```python
k, v = line.split('=', 1)
```

File: client/deis.py [hunk 2]
```diff
@@ -1302,6 +1302,19 @@ def config_unset(self, args):
         else:
             raise ResponseError(response)
 
+    def _read_config_from_path(self, path):
+        env_dict = {}
+        with open(path, 'r') as f:
+            for line in f.readlines():
```
Comment (mboersma) [inline client/deis.py]: Iterating directly on the [file object](https://docs.python.org/2/library/stdtypes.html#file.next) works too:
```python
for line in f:
```

File: client/deis.py (hunk 1 above)
Comment (eMxyzptlk) [inline client/deis.py]: done

File: client/deis.py (hunk 2 above)
Comment (eMxyzptlk) [inline client/deis.py]: done

File: client/deis.py [hunk 3]
```diff
@@ -1302,6 +1302,19 @@ def config_unset(self, args):
         else:
             raise ResponseError(response)
 
+    def _read_config_from_path(self, path):
+        env_dict = {}
+        with open(path, 'r') as f:
+            for line in f:
+                line = line.strip()
+                if re.match(r'.+=.+', line) is None:
+                    self._logger.warning('could not parse config line: "%s"', line)
```
Comment (bacongobbler) [inline client/deis.py]: If we have an empty line as explained in #3658, we should skip this warning. For example:

```
$ cat .env
FOO=bar

CAR=star
```

`could not parse config line: ""`

Comment (eMxyzptlk) [inline client/deis.py]: done

**Comment by bacongobbler** [comment]
Great! If you could now squish/amend your second and third commits to follow the style guide then I think we're good once Jenkins goes green.

**Comment by eMxyzptlk** [comment]
done :)

**Comment by bacongobbler** [comment]
code LGTM

**Comment by bacongobbler** [comment]
@kalbasit would you mind rebasing this? There was an issue with something upstream that caused Jenkins to fail: #3690

**Comment by kalbasit** [comment]
@bacongobbler done

**Comment by mboersma** [comment]
LGTM:
```console
$ cat .env
bla

$ deis config:push
could not parse config line: "bla"
Creating config... done, v4

=== yuppie-earthman
DEIS_APP: yuppie-earthman
FOO: bar
BAZ: 2
$ deis config:pull
could not parse config line: "bla"
$ cat .env 
FOO=bar
BAZ=2
```

**Pull request merged**